p2pd/main.go

```diff
@@ -98,13 +100,11 @@ func main() {
 
 		// if we explicitly specify a transport, we must also explicitly specify the listen addrs
```

Comment:
It seems like this comment is wrong, or the functionality doesn't match. It says when we specify a transport we must explicitly specify listen addrs, but then on the next line we make an assumption about the listen addresses for QUIC. Is this right?